### ISSUE TITLE
Collections are not correctly unmarshalled when @JsonbCreator is used

### ISSUE THREAD

**Issue by tosix1988** [eclipse-ee4j/yasson#230]
When using @JsonbCreator to annotate a factory method to create an instance of the type, collection properties of the type are not properly unmarshalled.

To demonstrate this issue, let's assume we have:

Some enum type:
```java
public enum EnumType {
	ONE,
	TWO;
}
```
Then we have two types, each having a collection of EnumType as a property.
EnumCollection (no @JsonbCreator annotation):
```java
public class EnumCollection {

	private List<EnumType> enums;
	
	public EnumCollection() {
	}
	
	public EnumCollection(List<EnumType> enums) {
		this.enums = enums;
	}
	
	public List<EnumType> getEnums() {
		return enums;
	}

	public void setEnums(List<EnumType> enums) {
		this.enums = enums;
	}
}
```
EnumCollectionCreator (using @JsonbCreator):
```java
public class EnumCollectionCreator {

	private List<EnumType> enums;
	
	private EnumCollectionCreator(List<EnumType> enums) {
		this.enums = enums;
	}
	
	@JsonbCreator
	public static EnumCollectionCreator create(@JsonbProperty("enums") List<EnumType> enums) {
		return new EnumCollectionCreator(enums);
	}

	public List<EnumType> getEnums() {
		return enums;
	}

	public void setEnums(List<EnumType> enums) {
		this.enums = enums;
	}
}
```
Now the following test passes for the EnumCollection type:
```java
public class EnumCollectionTest {

    @Test
    public void test() {
        EnumCollection coll = new EnumCollection(asList(EnumType.ONE, EnumType.TWO));
         
        Jsonb jsonb = JsonbBuilder.create();
        String result = jsonb.toJson(coll);
        
        assertEquals("{\"enums\":[\"ONE\",\"TWO\"]}", result);
        EnumCollection unmarshalled = jsonb.fromJson(result, EnumCollection.class);
        assertEquals(coll.getEnums(), unmarshalled.getEnums()); // collection of enums is property unmarshalled
    }
}
```
However, the following test fails for the type using @JsonbCreator (either the assertion fails, or accessing any element from the unmarshalled collection property throws ClassCastException):
```java
public class EnumCollectionCreatorTest {

    @Test
    public void test() {
        EnumCollectionCreator coll = EnumCollectionCreator.create(asList(EnumType.ONE, EnumType.TWO));
        
        Jsonb jsonb = JsonbBuilder.create();
        String result = jsonb.toJson(coll);
        assertEquals("{\"enums\":[\"ONE\",\"TWO\"]}", result);
        EnumCollectionCreator unmarshalled = jsonb.fromJson(result, EnumCollectionCreator.class); 
        
        // does not pass, coll.getEnums() is a list of enums, whereas unmarshalled.getEnums()
        // is a list of strings
        assertEquals(coll.getEnums(), unmarshalled.getEnums());
        
        // equivalently, the following line of code would throw ClassCastException when executed
        // EnumType t = unmarshalled.getEnums().get(0);
    }
}
```

Note: I have used Yasson 1.0.3 to reproduce. I can provide sample maven project if needed to reproduce.

**Comment by bravehorsie** [comment]
Thanks for reporting @tosix1988 !

**Comment by tosix1988** [comment]
Hi @bravehorsie, I have checked this issue again on master, and it seems that it has already been fixed in the meanwhile (most likely by 6ae0add15), so I believe this issue can be closed.

**Comment by aguibert** [comment]
closing out this issue since @tosix1988 indicated it has been fixed
